### PR TITLE
[CLT-1756] auto deploy docs site

### PR DESCRIPTION
**Pull request by BlackstoneEngineering**
This adds a github action script to auto deploy the docs site on changes to the master branch source files.

### CODE REVIEW

**Comment by BlackstoneEngineering** [comment]
@tung-vu-td - review?

File: .github/workflows/deploy-docs.yaml [hunk 1]
```diff
@@ -0,0 +1,62 @@
+name: Deploy Docs Microsite
+
+on:
+  workflow_dispatch:
+    inputs:
+      tags:
+        description: 'Who triggered this test run?'
```
Comment (tung-vu-td) [inline .github/workflows/deploy-docs.yaml:7]: Maybe just `run` instead of `test run`?

**Review by tung-vu-td — APPROVED** [review]
LGTM

**Comment by BlackstoneEngineering** [comment]
@tung-vu - LGTM?

**Pull request merged**